docs/source/guides/dockerfile_generation.rst

```diff
@@ -122,172 +120,229 @@ The following templates per distro are available for extending:
-
-   +---------------+------------------------------------------+
-   | Release type  | Description                              |
-   +===============+==========================================+
-   | base          | A base setup for all supported distros.  |
-   +---------------+------------------------------------------+
-   | cuda          | CUDA-supported templates.                |
-   +---------------+------------------------------------------+
-   | miniconda     | Conda-supported templates.               |
-   +---------------+------------------------------------------+
-   | python        | Python releases.                         |
-   +---------------+------------------------------------------+
++---------------+------------------------------------------+
+| Release type  | Description                              |
++===============+==========================================+
+| base          | A base setup for all supported distros.  |
++---------------+------------------------------------------+
+| cuda          | CUDA-supported templates.                |
++---------------+------------------------------------------+
+| miniconda     | Conda-supported templates.               |
++---------------+------------------------------------------+
+| python        | Python releases.                         |
++---------------+------------------------------------------+
 
-   where :code:`base_<distro>.j2` is extended from `base.j2 <https://github.com/bentoml/BentoML/tree/main/bentoml/_internal/bento/docker/templates/base.j2>`_
+where :code:`base_<distro>.j2` is extended from `base.j2 <https://github.com/bentoml/BentoML/tree/main/bentoml/_internal/bento/docker/templates/base.j2>`_
 
-   The templates hierarchy is as follows:
+The templates hierarchy is as follows:
 
-   .. code-block:: bash
+.. code-block:: bash
 
-       .
-       └── base.j2
-           ├── base_alpine.j2
-           │   ├── miniconda_alpine.j2
-           │   └── python_alpine.j2
-           ├── base_amazonlinux.j2
-           │   └── python_amazonlinux.j2
-           ├── base_debian.j2
-           │   ├── cuda_debian.j2
-           │   ├── miniconda_debian.j2
-           │   └── python_debian.j2
-           └── base_ubi8.j2
-               ├── cuda_ubi8.j2
-               └── python_ubi8.j2
+   .
+   └── base.j2
+       ├── base_alpine.j2
+       │   ├── miniconda_alpine.j2
+       │   └── python_alpine.j2
+       ├── base_amazonlinux.j2
+       │   └── python_amazonlinux.j2
+       ├── base_debian.j2
+       │   ├── cuda_debian.j2
+       │   ├── miniconda_debian.j2
+       │   └── python_debian.j2
+       └── base_ubi8.j2
+           ├── cuda_ubi8.j2
+           └── python_ubi8.j2
 
+Users can quickly imports any of the aforementioned templates for their
+customize ``Dockerfile.template``:
 
 .. code-block:: jinja
 
-    {% extends bento_auto_template %}
+    {% extends "miniconda_alpine.j2" %}
```

Comment:
when this template is used, the conda/cuda/distro options will be ignored, right?

Reply:
Wdym?

Reply:
when miniconda_alpine.j2 is used, it will always be a conda image without cuda, right?

what if user have the following config:

```yaml
docker:
   cuda_version: 11
   docker_template: ./my_template.j2 # with your example here
```

Reply:
oh yes, it would always be the conda image.

Reply:
I think we can add check for this pretty easily

Reply:
for now i can add this to the documentation.